src/lib/react-query/projects.ts

```diff
@@ -394,6 +395,44 @@ export function updateProjectSettingsMutationOptions({
 	>
 }
 
+export function changeMemberRoleMutationOptions({
+	projectApi,
+	projectId,
+	queryClient,
+}: {
+	projectApi: MapeoProjectApi
+	projectId: string
+	queryClient: QueryClient
+}) {
+	return {
+		...baseMutationOptions(),
+		mutationFn: async ({
+			deviceId,
+			roleId,
+		}: {
+			deviceId: string
+			roleId: MemberApi.RoleIdForNewInvite
```

Comment:
Think we should update core to expose the type that core uses for this method (`RoleIdAssignableToOthers`) and use that here. 

Depending on the tolerance for being blocked on core, could extract it manually in the meantime by doing `Parameters<MapeoProjectApi['$member']['assignRole']>[1]`.

Reply:
Created an issue in core: https://github.com/digidem/comapeo-core/issues/1108